src/llmq/dkgsessionmgr.cpp

```diff
@@ -395,11 +400,21 @@ bool CDKGSessionManager::GetVerifiedContributions(Consensus::LLMQType llmqType,
             ContributionsCacheKey cacheKey = {llmqType, pQuorumBaseBlockIndex->GetBlockHash(), proTxHash};
             auto it = contributionsCache.find(cacheKey);
             if (it == contributionsCache.end()) {
-                auto vvecPtr = std::make_shared<std::vector<CBLSPublicKey>>();
-                CBLSSecretKey skContribution;
-                if (!db->Read(std::make_tuple(DB_VVEC, llmqType, pQuorumBaseBlockIndex->GetBlockHash(), proTxHash), *vvecPtr)) {
+                CDataStream s(SER_DISK, CLIENT_VERSION);
+                if (!db->ReadDataStream(std::make_tuple(DB_VVEC, llmqType, pQuorumBaseBlockIndex->GetBlockHash(), proTxHash), s)) {
                     return false;
                 }
+                size_t vvec_size = ReadCompactSize(s);
+                CBLSPublicKey pubkey;
+                CBLSPublicKeyVersionWrapper pubkey_wrapped(pubkey, false);
+                std::vector<CBLSPublicKey> qv;
+                for ([[maybe_unused]] size_t _ : irange::range(vvec_size)) {
+                    s >> pubkey_wrapped;
+                    qv.emplace_back(pubkey_wrapped.Get());
```

Comment:
why not:

```suggestion
                CBLSPublicKey pubkey;
                std::vector<CBLSPublicKey> qv;
                for ([[maybe_unused]] size_t _ : irange::range(vvec_size)) {
                    s >> CBLSPublicKeyVersionWrapper(pubkey, false);
                    qv.emplace_back(pubkey);
```

`CBLSPublicKeyVersionWrapper` is a simple wrapper with one reference inside and one flag, it is not assumed to be a standalone variable with long life

Reply:
Good point! 👍 I missed the fact that it's indeed a reference 🙈